### PR TITLE
Add blog_id to analytics when available

### LINKED ISSUE #3371: Capture blog/site ID as a parameter in WPAnalytics calls where appropriate
Find all calls to event tracking in WPAnalytics around the app and pass in blog/site ID if it's a .com or a Jetpack site when it makes sense.

cc: @daniloercoli

### PR DESCRIPTION
**Pull request by daniloercoli**
Fix #3371

cc @astralbodies

### CODE REVIEW

**Comment by astralbodies** [comment]
Looks good from a Tracks perspective. Can't comment on code quality or app health so will need a dev to review that part. :wink:

File: WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java [hunk 1]
```diff
@@ -49,4 +58,102 @@ public static int getWordCount(String content) {
...
+     *
+     */
+    public static void trackWithBlogDetails(AnalyticsTracker.Stat stat, Blog blog, Map<String, Object> properties) {
+        if (blog == null || (!blog.isDotcomFlag() && !blog.isJetpackPowered())) {
+            AppLog.w(AppLog.T.STATS, "The passed blog obj is null or it's not a wpcom or Jetpack. Tracking analytics without blog info");
+            AnalyticsTracker.track(stat, properties);
+            return;
+        }
+
+        String blogID = StatsUtils.getBlogId(blog);
```
Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java]: We should probably move `StatsUtils.getBlogId` to another Util now it's used in other place than Stats. I'm also wondering if we should reuse it in other places where we need a remote blog id for .com/.org/jetpack site.

Comment (daniloercoli) [inline WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java]: > reuse it in other places where we need a remote blog id for .com/.org/jetpack site.

We should it everywhere in the app, where a remoteID is needed. 
Could we move this task to its own ticket?

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java]: Yep

Comment (maxme) [inline WordPress/src/main/java/org/wordpress/android/util/AnalyticsUtils.java]: New issue: https://github.com/wordpress-mobile/WordPress-Android/issues/3401

**Comment by maxme** [comment]
:shipit:

**Pull request merged**